src/utils.jl

```diff
@@ -169,6 +169,16 @@ for (type, _, _) in ABSTRACT_QUANTITY_TYPES
 end
 Base.keys(q::UnionAbstractQuantity) = keys(ustrip(q))
 
+# If atol specified in kwargs, validate its dimensions and then strip units
+function _strip_kws(dimcheck, kws)
+    if :atol in keys(kws)
```

Comment:
Should be faster to do `haskey(kws, :atol)`

Reply:
Updated.